python/cudaq/_cudaq.cpp

```diff
@@ -53,6 +56,19 @@ PYBIND11_MODULE(_pycudaq, mod) {
   mpiSubmodule.def(
       "initialize", []() { cudaq::mpi::initialize(); },
       "Initialize MPI if available.");
+  mpiSubmodule.def(
+      "rank", []() { return cudaq::mpi::rank(); }, "");
+  mpiSubmodule.def(
+      "num_ranks", []() { return cudaq::mpi::num_ranks(); }, "");
+  mpiSubmodule.def(
+      "all_gather",
+      [](std::size_t globalVectorSize, std::vector<double> &local) {
+        std::vector<double> global(globalVectorSize);
+        cudaq::mpi::all_gather(global, local);
+        return global;
+      },
+      "");
+  // mpiSubmodule.def("rank", []() { return cudaq::mpi::all_reduce(); }, "");
```

Comment:
```suggestion
```
rename member name and uncomment, or remove until we need it.

Reply:
this should be removed